packages/plugin-survey/src/index.ts

```diff
@@ -1,525 +1,178 @@
-          type: ParameterType.STRING,
-          pretty_name: "Placeholder",
-          default: "",
-        },
-        /** Text only: The number of rows (height) for the response text box. */
-        textbox_rows: {
-          type: ParameterType.INT,
-          pretty_name: "Textbox rows",
-          default: 1,
-        },
-        /** Text only: The number of columns (width) for the response text box. */
-        textbox_columns: {
-          type: ParameterType.INT,
-          pretty_name: "Textbox columns",
-          default: 40,
-        },
-        /**
-         * Text only: Type for the HTML <input> element.
-         * The `input_type` parameter must be one of "color", "date", "datetime-local", "email", "month", "number", "password", "range", "tel", "text", "time", "url", "week".
-         * If the `textbox_rows` parameter is larger than 1, the `input_type` parameter will be ignored.
-         * The `textbox_columns` parameter only affects questions with `input_type` "email", "password", "tel", "url", or "text".
-         */
-        input_type: {
-          type: ParameterType.SELECT,
-          pretty_name: "Input type",
-          default: "text",
-          options: [
-            "color",
-            "date",
-            "datetime-local",
-            "email",
-            "month",
-            "number",
-            "password",
-            "range",
-            "tel",
-            "text",
-            "time",
-            "url",
-            "week",
-          ],
-        },
-        /**
-         * All question types except HTML: value of the correct response. If specified, the response will be compared to this value,
-         * and an additional data property "correct" will store response accuracy (true or false).
-         */
-        correct_response: {
-          // TO DO: add correct response and accuracy scoring to data
-          type: ParameterType.STRING,
-          pretty_name: "Correct response",
-          default: null,
-        },
-      },
-    },
-    /** Whether or not to randomize the question order on each page */
-    randomize_question_order: {
-      type: ParameterType.BOOL,
-      pretty_name: "Randomize question order",
-      default: false,
-    },
-    /** Label of the button to move forward thorugh survey pages. */
-    button_label_next: {
-      type: ParameterType.STRING,
-      pretty_name: "Next button label",
-      default: "Next",
-    },
-    /** Label of the button to move backward through survey pages. */
-    button_label_back: {
-      type: ParameterType.STRING,
-      pretty_name: "Back button label",
-      default: "Back",
-    },
-    /** Label of the button to submit responses. */
-    button_label_finish: {
+    /**
+     * A SurveyJS survey model defined in JSON.
+     * See: https://surveyjs.io/form-library/documentation/design-survey/create-a-simple-survey#define-a-static-survey-model-in-json
+     */
+    survey_json: {
       type: ParameterType.STRING,
```

Comment:
What's the motivation for serialization here instead of directly using the original object?

Reply:
Hmm. I think I did this to make it clear to users that `survey_json` had to be valid JSON so that they didn't include any functions inside the object. But I just tried this with functions nested in the `survey_json` object, and it works 😮  I guess this is because `replaceFunctionsWithValues` is recursive and can therefore handle functions that are nested arbitrarily in an object - I didn't realize this was the case.
Thanks for the suggestion - this is much better. I will change this to:
```javascript
type: ParameterType.OBJECT
```
and update the docs and examples. And since any functions should be called before the plugin's trial method, I can check for valid JSON in the plugin code. Does that sound ok?

Reply:
@jodeleeuw @bjoluc if you're both happy with this change (`survey_json` takes JS objects rather than JSON strings), then my next question is: should I change the name of the `survey_json` parameter? 
I think I'm in favor of sticking with `survey_json` for consistency with the SurveyJS docs, but I could add a note in the docs explaining this. 
Another option might be to call the parameter `survey_object` instead of `survey_json`.

Reply:
>  But I just tried this with functions nested in the survey_json object, and it works 😮

Yup, but I think we shouldn't embrace this as it will break in v8 where only plugin-defined (nested object) parameters are evaluated.

> And since any functions should be called before the plugin's trial method, I can check for valid JSON in the plugin code.

I don't know if this is necessary at all: Doesn't SurveyJS validate the JSON object already? And if anything non-JSON slips through that SurveyJS doesn't complain about, why should our plugin complain :smile: 

> I think I'm in favor of sticking with survey_json for consistency with the SurveyJS docs, but I could add a note in the docs explaining this.

Me too – I think it's clear from their docs and our examples then that `survey_json` means "survey JSON object" not "survey JSON string". `survey_object` could also refer to a SurveyJS `Survey` object, but we could of course use `survey_json_object` if you think that's clearer than `survey_json`. I'm fine with both.

Reply:
> Yup, but I think we shouldn't embrace this as it will break in v8 where only plugin-defined (nested object) parameters are evaluated.

Aha, thanks for flagging this! Just to clarify - are you saying that, in v8, the whole `survey_json` parameter can still be a function that will be evaluated by jsPsych? But values nested _within_ that object won't be evaluated? I agree about not highlighting an approach that will break in v8, so I just want to be sure I understand what you mean before modifying the docs/examples.

Everything else makes sense. Thanks very much for your input!

Reply:
Sorry for the late reply!

> are you saying that, in v8, the whole survey_json parameter can still be a function that will be evaluated by jsPsych? But values nested within that object won't be evaluated?

:100: :+1: